File: src/Entity/CalendarSetting.php
<?php

namespace App\Entity;

use App\Repository\CalendarSettingRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: CalendarSettingRepository::class)]
class CalendarSetting
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column]
    private ?int $slotDuration = 30;

    #[ORM\Column(type: Types::TIME_MUTABLE, nullable: true)]
    private ?\DateTimeInterface $pauseStart = null;

    #[ORM\Column(type: Types::TIME_MUTABLE, nullable: true)]
    private ?\DateTimeInterface $pauseEnd = null;

    #[ORM\Column]
    private ?int $doctorId = 1;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getSlotDuration(): ?int
    {
        return $this->slotDuration;
    }

    public function setSlotDuration(int $slotDuration): static
    {
        $this->slotDuration = $slotDuration;

        return $this;
    }

    public function getPauseStart(): ?\DateTimeInterface
    {
        return $this->pauseStart;
    }

    public function setPauseStart(?\DateTimeInterface $pauseStart): static
    {
        $this->pauseStart = $pauseStart;

        return $this;
    }

    public function getPauseEnd(): ?\DateTimeInterface
    {
        return $this->pauseEnd;
    }

    public function setPauseEnd(?\DateTimeInterface $pauseEnd): static
    {
        $this->pauseEnd = $pauseEnd;

        return $this;
    }

    public function getDoctorId(): ?int
    {
        return $this->doctorId;
    }

    public function setDoctorId(int $doctorId): static
    {
        $this->doctorId = $doctorId;

        return $this;
    }
}

File: doctor-calendar/src/main/java/com/tabibnet/calendar/model/CalendarSetting.java
package com.tabibnet.calendar.model;

import java.time.LocalTime;

public class CalendarSetting {
    private int id;
    private int slotDuration;
    private LocalTime pauseStart;
    private LocalTime pauseEnd;
    private int doctorId;

    public CalendarSetting() {
    }

    public CalendarSetting(int id, int slotDuration, LocalTime pauseStart, LocalTime pauseEnd, int doctorId) {
        this.id = id;
        this.slotDuration = slotDuration;
        this.pauseStart = pauseStart;
        this.pauseEnd = pauseEnd;
        this.doctorId = doctorId;
    }

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public int getSlotDuration() { return slotDuration; }
    public void setSlotDuration(int slotDuration) { this.slotDuration = slotDuration; }

    public LocalTime getPauseStart() { return pauseStart; }
    public void setPauseStart(LocalTime pauseStart) { this.pauseStart = pauseStart; }

    public LocalTime getPauseEnd() { return pauseEnd; }
    public void setPauseEnd(LocalTime pauseEnd) { this.pauseEnd = pauseEnd; }

    public int getDoctorId() { return doctorId; }
    public void setDoctorId(int doctorId) { this.doctorId = doctorId; }
}
